zipkin-lens/src/components/TraceViewer/index.js

```diff
@@ -0,0 +1,163 @@
+import React from 'react';
+import Dropzone from 'react-dropzone';
+
+import DetailedTraceSummary from '../DetailedTraceSummary';
+import { treeCorrectedForClockSkew, detailedTraceSummary } from '../../zipkin';
+
+const dropzoneBaseStyle = {
+  width: 200,
+  height: 60,
+  fontSize: '18px',
+  padding: '8px',
+  borderWidth: 2,
+  borderColor: '#666',
+  borderStyle: 'dashed',
+  borderRadius: 5,
+};
+
+const dropzoneActiveStyle = {
+  borderStyle: 'solid',
+  borderColor: '#6c6',
+  backgroundColor: '#eee',
+};
+
+class TraceViewer extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      startTs: null,
+      endTs: null,
+      traceSummary: null,
+      cannotOpenFile: false,
+      isInvalidFormat: false,
+    };
+    this.handleDrop = this.handleDrop.bind(this);
+    this.handleCancel = this.handleCancel.bind(this);
+    this.handleStartAndEndTsChange = this.handleStartAndEndTsChange.bind(this);
+  }
+
+  handleDrop(files) {
+    const [file] = files;
+    const fileReader = new FileReader();
+
+    fileReader.onload = () => {
+      const { result } = fileReader;
+      let rawTrace;
+      try {
+        rawTrace = JSON.parse(result);
+      } catch (error) {
+        this.setState({
+          traceSummary: null,
+          cannotOpenFile: false,
+          isInvalidFormat: true,
+        });
+      }
+      const traceSummary = detailedTraceSummary(
+        treeCorrectedForClockSkew(rawTrace),
+      );
+
+      this.setState({
+        startTs: 0,
+        endTs: traceSummary.duration,
```

Comment:
`startTs` and `endTs` must be set when `traceSummary` is changed.
If not do so, MiniTimeline's range will be bad.